commit 40e496dede06242a0299b0af0acdbb39d625523c (refs/changes/84/2484/2)
Author: David Pacheco <dap@joyent.com>
Date:   2017-09-01T11:28:29-07:00 (2 years, 1 month ago)
    
    joyent/mdb_v8#94 publish script use of mdb should ignore mdbrc file
    joyent/mdb_v8#95 tst.jsclosure.js breaks with long workspace path names
    Reviewed by: Tim Kordas <tim.kordas@joyent.com>
    Approved by: Tim Kordas <tim.kordas@joyent.com>

diff --git a/test/standalone/tst.jsclosure.js b/test/standalone/tst.jsclosure.js
index 7919001..da5ab26 100644
--- a/test/standalone/tst.jsclosure.js
+++ b/test/standalone/tst.jsclosure.js
@@ -108,7 +108,11 @@ gcore.on('exit', function (code) {
 		console.log('mdb stderr: ' + data);
 	});
 
-	var mod = util.format('::load %s\n', common.dmodpath());
+	/*
+	 * The '1000$w' sets the terminal width to a large value to keep MDB
+	 * from inserting newlines at the default 80 columns.
+	 */
+	var mod = util.format('1000$w; ::load %s\n', common.dmodpath());
 	doCmd(mod);
 });
 
diff --git a/tools/publish b/tools/publish
index a7e205d..7d6d8bd 100755
--- a/tools/publish
+++ b/tools/publish
@@ -153,7 +153,7 @@ function pub_sanity_check_file
 	tmpfile=/var/tmp/mdbv8publish.$$
 	rm -f /var/tmp/mdbv8publish.$$
 	echo -n "Checking for release tag on "$file" ... "
-	mdb -e '::load '"$file"'; !touch '"$tmpfile"'; ! sleep 300' \
+	mdb -S -e '::load '"$file"'; !touch '"$tmpfile"'; ! sleep 300' \
 	    "$file" > /dev/null 2>&1 &
 	pid="$!"
 	for (( i = 0; i < 30; i++ )) {
@@ -165,7 +165,7 @@ function pub_sanity_check_file
 	}
 	[[ -f "$tmpfile" ]] || fail "failed"
 	rm -f "$tmpfile"
-	tag="$(mdb -p "$pid" -e '*mdbv8_vers_tag/s' | awk '{print $2}' |
+	tag="$(mdb -S -p "$pid" -e '*mdbv8_vers_tag/s' | awk '{print $2}' |
 	    sed -e s'#,##g')"
 	kill "$pid" > /dev/null 2>&1
 	echo "\"$tag\""
